stylesheets/_component.toggle-switch.scss

```diff
@@ -0,0 +1,84 @@
+.toggle-switch {
+    @extend %hide;
+
+    &:checked + .toggle-switch-label {
+        background-color: color(success); //#1ABC9C
+        border-color: transparent;
+        color: color(white);
+
+        &:before {
+            left: 50%;
+            right: 0%;
+        }
+    }
+
+    &:indeterminate + .toggle-switch-label {
+        background-color: color(warning);
+
+        &:before {
+            left: 25%;
+            right: 25%;
+        }
+
+        &:after {
+            color: color(white);
+            opacity: 0;
+        }
+    }
+
+    &.toggle-switch--large + .toggle-switch-label {
+        height: 2em;
+        width: 4em;
+
+        &:after {
+            font-size: 32px;
+            left: 15px;
+            letter-spacing: 32px;
+        }
+    }
+}
+
+.toggle-switch-label {
+    background-clip: padding-box;
+    background-color: color(grey, light);
+    border-color: transparent;
+    border-radius: 1.5em;
```

Comment:
Would a percentage-based unit be better here, incase the height ever changed (for whatever reason)?